### ISSUE TITLE
no hardware decoding using MPV on egpu

### ISSUE THREAD

**Issue by ptrs2010** [goalque/automate-eGPU#35]
I can not get hardware decoding to work in MPV with a Zotac GTX 970 in an Akitio Thunder2 box on OSX 10.10.5 Yosemite (tested both a Mac Mini late 2012 and Macbook Pro mid 2012). This is while trying to play a standard “older” H264 file, which should be easily supported. Hardware decoding on the Intel igpu works fine, however when I try to play the files using the egpu, videotoolbox has trouble initiating and MPV falls back to software decoding. 

I have tried multiple files in both .mp4 and .mkv container format, all without success. Also a lengthy google search did not reveal an answer. I tried the pre-compiled MPV, as well as a self-compiled —HEAD version using homebrew. does videotoolbox only support natively supported cards? This would exclude the entire Maxwell series from using gpu decoding (and will spell more trouble for much- needed Pascal-based h265 decoding when Nvidia will get around to release their new web-driver!).

I run the latest homebrew-compiled mpv as follows:

>mpv --no-config -v --hwdec=auto h264_file.mkv

I disconnect the HDMI on the Mac Mini to ensure OpenGL is loaded for Nvidia:

[vo/opengl] GL_VENDOR='NVIDIA Corporation'
[vo/opengl] GL_RENDERER='NVIDIA GeForce GTX 970 OpenGL Engine'
[vo/opengl] GL_SHADING_LANGUAGE_VERSION='4.10'

However the result is always:

IOVARendererID property not found 
[12:30:54.700] vtDecompressionDuctCreate signalled err=-8973 (err) (Could not select and open decoder instance) at /SourceCache/CoreMedia_frameworks/CoreMedia-1562.240/Sources/VideoToolbox/VTDecompressionSession.c line 1181
[vd] failed to init videotoolbox decoder: -1313558101

Although the first line is suspect I think it is just a warning, as it also appears when playing on the igpu (on which hardware decoding works fine), so the problem appears to me to be videotoolbox not wanting to do hardware decoding on the 970. 

Sorry for the lengthy email, but I am really stuck here, so any help with this would be greatly appreciated!

**Comment by leavittx** [comment]
Yes, I think I have the same error (code -1313558101 after calling av_videotoolbox_default_init2).
Nvidia GTX 980, OSX 10.11.6, latest Web Driver 346.03.15f03

**Comment by ptrs2010** [comment]
After some more experimenting on an iMac late 2012, the MBP mid 2012 with dgpu and a mac mini mid 2012 I found that the problem probably is due to mpv/ffmpeg not handling the dual/triple graphics card configuration properly. On iMac with onky Nividia gpu, the hardware acceleration works fine and on the MBP with the integrated graphics disabled it works fine too (despite/because of two Nvidia cards being active at the same time). However my intention is to use the egpu with the mac mini. Does anyone know of a way to disable the integrated graphics on the imac (preferably during boot)?
